zip-codes/scripts/search.js

```diff
@@ -2,8 +2,120 @@
 
 (function(module) {
 
+  var placeHolder = function() {
+    $('input[placeholder]').each(function () {
+      $(this).attr('size', $(this).attr('placeholder').length);
+    });
+  };
+
+  var stateFilter = function() {
+    webDB.execute(
+      'SELECT DISTINCT state ' +
+      'FROM zips ' +
+      'ORDER BY state ASC',
+      function(states) {
+        states.forEach(function(ele) {
+          $('#state-select').append('<option value="' + ele.state + '">' + ele.state + '</option>');
+        });
+      }
+    );
+  };
+
+  var cityFilter = function() {
+    $('#state-select').on('change', function(){
+      document.getElementById('city-select').options.length = 1;
+      webDB.execute(
+        'SELECT DISTINCT city, latitude, longitude ' +
+        'FROM zips ' +
+        'WHERE state = "' + $(this).val() + '" ORDER BY city ASC',
+        function(cities) {
+          cities.forEach(function(ele) {
+            if($('#city-select option[value="' + ele.city + '"]').length === 0) {
+              $('#city-select').append('<option value="' + ele.city + '">' + ele.city + '</option>');
```

Comment:
I assume you are trying to eliminate duplicates here. If these duplicates are due to changing states and double appending, you could instead simply clear our your select box before you append cities using `$('#city-select').empty()`.